### PR TITLE
Documentation for Any

### PR DESCRIPTION
**Pull request by oldergod**

### CODE REVIEW

File: wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt [hunk 1]
```diff
@@ -91,7 +93,7 @@ class AnyMessage(
       }
 
       override fun decode(reader: ProtoReader): AnyMessage {
-        var typeUrl = "square.github.io/wire/unknown" // TODO: what does protoc do?
+        var typeUrl = ""
```
Comment (oldergod) [inline wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt:96]: protoc does that.

File: wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt [hunk 2]
```diff
@@ -19,8 +19,10 @@ import okio.ByteString
 import kotlin.jvm.JvmField
 
 /**
- * TODO(oldergod): documentation
- * TODO(oldergod): JSON
+ * A [ProtoAdapter] for the `google.protobuf.Any` type. `Any` wraps an arbitrary protobuf message.
```
Comment (swankjesse) [inline wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt:22]: Wanna paste a usage example from the test suite?

File: wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt (hunk 1 above)
Comment (swankjesse) [reply wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt:96]: SGTM!

Comment (swankjesse) [reply wire-library/wire-runtime/src/commonMain/kotlin/com/squareup/wire/AnyMessage.kt:96]: (though it’s now clear we don’t have a test for this)

File: wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt [hunk 3]
```diff
@@ -65,5 +70,74 @@ class AnyMessageTest {
...
+      fail()
+    } catch (expected: IllegalStateException) {
+      assertEquals("recompile class com.squareup.wire.AnyMessageTest.NoTypeUrlMessage to use " +
+          "it with AnyMessage",
+          // We cover all platform cases.
+          expected.message!!.replace("\$NoTypeUrlMessage", ".NoTypeUrlMessage"))
+    }
+  }
+
+  private class NoTypeUrlMessage(
```
Comment (swankjesse) [inline wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt:121]: Nice!

File: wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt [hunk 4]
```diff
@@ -65,5 +70,74 @@ class AnyMessageTest {
...
+  }
+
+  @Test fun packWithoutATypeUrlThrows() {
+    try {
+      AnyMessage.pack(NoTypeUrlMessage())
+      fail()
+    } catch (expected: IllegalStateException) {
+      assertEquals("recompile class com.squareup.wire.AnyMessageTest.NoTypeUrlMessage to use " +
+          "it with AnyMessage",
+          // We cover all platform cases.
```
Comment (swankjesse) [inline wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt:116]: I don’t follow this replace . . .  why not change the expectation?

Comment (oldergod) [reply wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt:116]: In JVM, we get `com.squareup.wire.AnyMessageTest$NoTypeUrlMessage`, we native we get `com.squareup.wire.AnyMessageTest.NoTypeUrlMessage`

Comment (oldergod) [reply wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt:116]: and JS has `NoTypeUrlMessage` only without the package name...

Comment (oldergod) [reply wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt]: Went with `assertTrue(expected.message!!.endsWith("NoTypeUrlMessage to use it with AnyMessage"))`

Comment (swankjesse) [reply wire-library/wire-tests/src/commonTest/kotlin/com/squareup/wire/AnyMessageTest.kt]: multiplatform was the part I hadn’t anticipated!

**Pull request merged**